### ISSUE TITLE
random sign flips after RGEs

### ISSUE THREAD

**Issue by xmarcano** [wilson-eft/wilson#63]
We are using the wilson.match_run function to obtain the low-energy values of some Higgs-lepton and 4-lepton WCs, however we do not understand the sign of the output WCs. They seem random and don't make much sense to us. 

For example, starting with phil1_13=+1 at 1TeV and running it a bit down to 999GeV, we obtain phil1_13~+0.9997. Nevertheless, if we start with phil1_13=-1, we get again +0.9997, when it should be negative. On the other hand, doing the same for phil1_23 gives a consistent sign after running, while phil1_12 flips the sign in both cases.

We have also tried it for other WCs, including also WCs that are zero at high-scale, and again the sign seems random and dependent on the flavor. We also found that the sign of a given WC may change depending on the final scale on is running to. I attach a pdf with some of our trials. 

Are we understanding something wrong or could it be a numerical instability when solving the RGEs? 

We have cross-checked that the absolute value of the run WCs agrees with that of the DsixTools package, so it won't have any effect in many observables. Nevertheless we are worried about possible interferences when several WCs play a role. 

Thanks!

[RGE_tests.pdf](https://github.com/wilson-eft/wilson/files/5957755/RGE_tests.pdf)

**Comment by DavidMStraub** [comment]
Please paste your code as code block, not as PDF. This makes it impossible to copy & paste.

**Comment by xmarcano** [comment]
sure, here it goes the part I was mentioning:

e-tau sector (always + sign from RGEs):
```
In[]:
HIGHscale=1000
LOWscale=999
WChigh=1
WC1='phil1_13'

wc = Wilson({WC1:WChigh/HIGHscale/HIGHscale}, scale=HIGHscale, eft='SMEFT', basis='Warsaw')
print('At mu=',HIGHscale,'GeV: ', WC1,'=',WChigh)

WCrun = wc.match_run(scale=LOWscale, eft='SMEFT', basis='Warsaw')
print('At mu=',LOWscale,'GeV: ', WC1,'=',np.real(WCrun[WC1])*HIGHscale*HIGHscale)

wc = Wilson({WC1:-WChigh/HIGHscale/HIGHscale}, scale=HIGHscale, eft='SMEFT', basis='Warsaw')
print('At mu=',HIGHscale,'GeV: ', WC1,'=',-WChigh)

WCrun = wc.match_run(scale=LOWscale, eft='SMEFT', basis='Warsaw')
print('At mu=',LOWscale,'GeV: ', WC1,'=',np.real(WCrun[WC1])*HIGHscale*HIGHscale)
```
```
Out[]:
At mu= 1000 GeV:  phil1_13 = 1
At mu= 999 GeV:  phil1_13 = 0.9999712919291377
At mu= 1000 GeV:  phil1_13 = -1
At mu= 999 GeV:  phil1_13 = 0.9999712916429004
```

mu-tau sector (RGEs preserve sign):
```
In[]:
HIGHscale=1000
LOWscale=999
WChigh=1
WC1='phil1_23'

wc = Wilson({WC1:WChigh/HIGHscale/HIGHscale}, scale=HIGHscale, eft='SMEFT', basis='Warsaw')
print('At mu=',HIGHscale,'GeV: ', WC1,'=',WChigh)

WCrun = wc.match_run(scale=LOWscale, eft='SMEFT', basis='Warsaw')
print('At mu=',LOWscale,'GeV: ', WC1,'=',np.real(WCrun[WC1])*HIGHscale*HIGHscale)

wc = Wilson({WC1:-WChigh/HIGHscale/HIGHscale}, scale=HIGHscale, eft='SMEFT', basis='Warsaw')
print('At mu=',HIGHscale,'GeV: ', WC1,'=',-WChigh)

WCrun = wc.match_run(scale=LOWscale, eft='SMEFT', basis='Warsaw')
print('At mu=',LOWscale,'GeV: ', WC1,'=',np.real(WCrun[WC1])*HIGHscale*HIGHscale)
```
```
Out[]:
At mu= 1000 GeV:  phil1_23 = 1
At mu= 999 GeV:  phil1_23 = 0.9999712906551955
At mu= 1000 GeV:  phil1_23 = -1
At mu= 999 GeV:  phil1_23 = -0.9999712909455126
```

e-mu sector (RGEs flip sign):
```
In[]:
HIGHscale=1000
LOWscale=999
WChigh=1
WC1='phil1_12'

wc = Wilson({WC1:WChigh/HIGHscale/HIGHscale}, scale=HIGHscale, eft='SMEFT', basis='Warsaw')
print('At mu=',HIGHscale,'GeV: ', WC1,'=',WChigh)

WCrun = wc.match_run(scale=LOWscale, eft='SMEFT', basis='Warsaw')
print('At mu=',LOWscale,'GeV: ', WC1,'=',np.real(WCrun[WC1])*HIGHscale*HIGHscale)

wc = Wilson({WC1:-WChigh/HIGHscale/HIGHscale}, scale=HIGHscale, eft='SMEFT', basis='Warsaw')
print('At mu=',HIGHscale,'GeV: ', WC1,'=',-WChigh)

WCrun = wc.match_run(scale=LOWscale, eft='SMEFT', basis='Warsaw')
print('At mu=',LOWscale,'GeV: ', WC1,'=',np.real(WCrun[WC1])*HIGHscale*HIGHscale)
```
```
Out[]:
At mu= 1000 GeV:  phil1_12 = 1
At mu= 999 GeV:  phil1_12 = -0.9999712929225841
At mu= 1000 GeV:  phil1_12 = -1
At mu= 999 GeV:  phil1_12 = 0.9999712929225841
```

**Comment by DavidMStraub** [comment]
Thanks.

So, the answer is: yes, the sign is essentially random, but unphysical.

What happens is that, after running, the fermion masses need to be rediagonalized to go the SMEFT Warsaw basis as defined in WCxf (see the WCxf paper for details). The code for it is here:

https://github.com/wilson-eft/wilson/blob/master/wilson/run/smeft/classes.py#L125-L142

Now, what happens is that the matrices `UeL` and `UeR` can e.g. have the approximate form `diag(1, -1, 1)`. There's nothing wrong with it, but it flips the sign of the muon fields. Which signs are chosen for the diagonal elements is essentially random, as the RGE induce extremely tiny off-diagonal elements in the lepton mass matrix.

So what you are observing here is the fact that the basis before and after running is different, due to the way WCxf defines the Warsaw basis, but it is still perfectly consistent - unless you start combining outputs from several runs with different inputs which correspond to different basis, but there's obviously no reason to do that.

To summarize, I think it's not a bug.

**Comment by DavidMStraub** [comment]
I could be wrong of course, so curious to hear your opinion :slightly_smiling_face:

**Comment by xmarcano** [comment]
Hi David, thanks for the detailed explanation. 

If the sign comes from the rediagonalization, I agree with you. It should be unphysical and consistent even if several WCs interfere for a given observable. 

Thanks!
